api/app/GraphQL/Directives/HasPermissionDirective.php

```diff
@@ -40,23 +39,29 @@ public function handleField(FieldValue $fieldValue, Closure $next): FieldValue
         $originalResolver = $fieldValue->getResolver();
 
         $fieldValue->setResolver(function ($root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo) use ($originalResolver) {
-            $requiredRoles = $this->directiveArgValue('requiredRoles');
+            $permissions = $this->directiveArgValue('permissions');
             $ownerKey = $this->directiveArgValue('ownerKey');
 
             // Throw in case of an invalid schema definition to remind the developer
-            if ($requiredRoles === null || empty($requiredRoles)) {
-                throw new DefinitionException("Missing argument 'requiredRoles' for directive '@canAccess'.");
+            if ($permissions === null || empty($permissions)) {
+                throw new DefinitionException("Missing argument 'permissions' for directive '@hasPermission'.");
             }
 
             $user = User::find($context->user()->id);
             $owned = $ownerKey && $user->id === $root->$ownerKey;
-            $authorized = $user && $user->hasRole($requiredRoles);
+            $checkOwnership = false;
 
-            if ($owned || $authorized) {
-                return $originalResolver($root, $args, $context, $resolveInfo);
+            foreach ($permissions as $permission) {
+                if (strpos('-own-', $permission)) {
+                    $checkOwnership = true;
+                    break;
+                }
```

Comment:
We only require the user to have one of the permissions. We have to check for ownership along with the 'own' permission specifically. Right now we might require ownership even if the user has the 'view-all' permission.